### PR TITLE
Auto typecheck

### PR DESCRIPTION
**Pull request by Seanny123**
Added more static checks and updated README.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,12 +4,14 @@ before_install:
...
   - pip install -U pytest coverage pytest-cov
-  - pip install codecov black
+  - pip install codecov black mypy pylint
 
 install:
   - pip install -e . # install package
 
 script:
+  - pylint --rcfile=setup.cfg PeptideBuilder/
+  - mypy PeptideBuilder/
```
Comment (clauswilke) [inline .travis.yml]: For both the `pylint` and the `mypy` lines, could you add a brief comment stating what these lines do? Just like the next two lines, `black`, and `pytest`.

File: README.md [hunk 2]
```diff
@@ -32,12 +32,17 @@ To test whether your installation works properly, run `pytest` in the top-level
 
 ## Contributing
 
-Pull requests are welcome on GitHub. However, please note contributions must be styled with [`black`](https://black.readthedocs.io/en/stable/) and pass the `pytest` unit tests to be accepted.
+Pull requests are welcome on GitHub. However, please note contributions must be
+ styled with [`black`](https://black.readthedocs.io/en/stable/), linted with
+ `pylint`, type-checked with `mypy` and pass the `pytest` unit tests to be
```
Comment (clauswilke) [inline README.md]: Please place a comma after `mypy` (Oxford comma rule).

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+python_version = 3.7
+
+[mypy-Bio.*]
+ignore_missing_imports = True
+
+[mypy-numpy]
+ignore_missing_imports = True
+
+[pylint]
+disable =
```
Comment (clauswilke) [inline setup.cfg]: There is a lot of stuff under `disable`. Could you add a comment or two explaining what things (broadly speaking) are being disabled or enabled?

**Comment by Seanny123** [comment]
Made all the changes requested. Let me know what you think.

**Comment by clauswilke** [comment]
Thanks! @mtien and I will take a stab at resolving issue #8, and then I'll release.

**Pull request merged**